logstash-core/lib/logstash/pipeline.rb

```diff
@@ -415,10 +430,13 @@ def flush_filters(options = {}, &block)
   end
 
   def start_flusher
+    # Invariant to help detect improper initialization
+    raise "Attempted to start flusher on a stopped pipeline!" if stopped?
```

Comment:
is there a scenario where we predict this to happen? `start_flusher` is called immediately after making `@running` true.